modules/ml/src/main/java/org/apache/ignite/ml/preprocessing/encoding/EncoderTrainer.java

```diff
@@ -343,6 +441,81 @@ else if (lbVal instanceof Double)
         return categoryFrequencies;
     }
 
+    /**
+     * Updates frequencies by values and features.
+     *
+     * @param row Feature vector.
+     * @param targetCounters Holds the frequencies of categories by values and features.
+     * @return target counter.
+     */
+    private TargetCounter[] updateTargetCountersForNextRow(LabeledVector row,
+                                                           TargetCounter[] targetCounters) {
+        if (targetCounters == null)
+            targetCounters = initializeTargetCounters(row);
+        else
+            assert targetCounters.length == row.size() : "Base preprocessor must return exactly "
+                + targetCounters.length + " features";
+
+        double targetValue = row.features().get(targetLabelIndex);
+
+        for (int i = 0; i < targetCounters.length; i++) {
+            if (handledIndices.contains(i)) {
+                String strVal;
+                Object featureVal = row.features().getRaw(i);
+
+                if (featureVal.equals(Double.NaN)) {
+                    strVal = EncoderPreprocessor.KEY_FOR_NULL_VALUES;
+                    row.features().setRaw(i, strVal);
+                }
+                else if (featureVal instanceof String)
+                    strVal = (String)featureVal;
+                else if (featureVal instanceof Double)
```

Comment:
Maybe add type conversion to Doulbe from another Number types (and boolean)

Reply:
Add `Number` & `Boolean`